hal-core/src/mem/mod.rs

```diff
@@ -67,17 +67,31 @@ impl<A: Address> Region<A> {
         addr > self.base && addr < self.end_addr()
     }
 
-    /// Returns the end address of the memory region.
+    /// Returns the last address (inclusive) of the memory region.
     pub fn end_addr(&self) -> A {
-        self.base + self.size
+        self.base + (self.size - 1)
     }
```

Comment:
Why make this inclusive? one-past-the-end seems like a more common and convenient "end" value to be working with

Reply:
i could go either way — honestly the goal was just to make the `contains` check not be broken, but we could just change it there. to me, inclusive feels more intuitive, but i could be convinced otherwise. @iximeow  wanna bikeshed this?

Reply:
changed everything to inclusive in 7687749